### PR TITLE
Fix searching in Info-mode

### PR DESCRIPTION
**Pull request by DevJac**
This fixes some of the bugs with Info-mode searching. There is still a problem with search direction when using isearch (rather than evil-search).

### CODE REVIEW

**Comment by DevJac** [comment]
I made the following observation about the previous code in #281:

> One oddity I noticed in the below lines, is that ? is hard-coded to use evil-ex-search-backwards, which means ? is hard coded to behave as though Evil search is enabled. Meanwhile n and N try to be dynamic.

**Comment by Ambrevar** [comment]
Good catch!  I suggested a possibly better fix, let me know what you think.  Cheers!

**Comment by DevJac** [comment]
Where can I see your suggestion @Ambrevar ?

**Comment by Ambrevar** [comment]
Sorry, forgot to click the "submit" button.  You should see it now.

File: modes/info/evil-collection-info.el [hunk 1]
```diff
@@ -98,7 +103,9 @@
     "gk" 'Info-prev
 
     "g?" 'Info-summary
-    "?" 'evil-ex-search-backward        ; Else this would be `Info-summary'.
+    "?" (if (evil-collection-evil-search-enabled)  ; Else this would be `Info-summary'.
```
[the remark this answers is not in the document]
Comment (DevJac) [reply modes/info/evil-collection-info.el]: You're probably right. That is a variable equal to the following:
```
(menu-item "" nil :filter
	   (lambda
	     (&optional _)
	     (if
		 (eq evil-search-module 'evil-search)
		 #'evil-ex-search-backward #'evil-search-backward)))
```
While were on the topic, do you have any idea what all that is for?

**Comment by DevJac** [comment]
There's a decision to be made here, with regards to narrowed buffers and how evil search should handle them in Info-mode. In simplified form, the question is: Should we search the entire manual or just the current page?

Our options are:
1) Use isearch, which is what I'm doing here, which will search the entire manual. However, this breaks the forward/backward (n/N) behavior after doing a backwards search (?). The reason I'm doing this now is because it matches what was being done previously.

2) Use evil search functions (I'm not talking about the evil-search-module here, just evil search related functions in general) which will behave correctly, but will not search only the narrowed buffer. In other words, it will not search the entire manual. I like this option better personally, because it may not be as well featured, but it does use evil correctly and consistently.

**Comment by Ambrevar** [comment]
I think this is not really a question of "Evil or not", but rather an "Info" specificity.
I'd opt for having `/` and `?` and `n` and `N` restricted to the current page, not the entire manualy.  If the user wants to search the entire manual, they can always use `C-s`.

File: modes/info/evil-collection-info.el (hunk 1 above)
[the remark this answers is not in the document]
Comment (Ambrevar) [reply modes/info/evil-collection-info.el]: > what all that is for?

What do you mean?  You mean the `menu-item` business?
If I'm not mistaken, it's an Evil hack used to implement `operator` features.  Sorry, I don't know much more about this.

**Comment by DevJac** [comment]
Ok. I made the suggested changes.

This will no longer search the entire manual like the raw isearch calls would, but this way is more consistent since it fixes the search direction bug the raw isearch calls caused.

**Comment by Ambrevar** [comment]
Thanks!

**Pull request merged**